docs/releases/2.17.md

```diff
@@ -14,7 +14,7 @@ depth: 1
 Here are other changes related to the redesign:
 
  * Switch the Wagtail branding font to a system font stack (Steven Steinwand)
-
+ * Migrate editor's guide to markdown(MyST) instead of RST (Shwet Khatri)
```

Comment:
as noted above, no need to update the release notes, the final release that this ends up in may change and also this will cause PR conflicts if other things are merged in before this PR.